### ISSUE TITLE
Ensure that soil biosamples have two depth values (minimum and maximum)

### ISSUE THREAD

**Issue by turbomam** [microbiomedata/issues#17]
- `depth2` is being deprecated in https://github.com/microbiomedata/nmdc-schema/blob/issue-486-data-to-7-0
- the released [depth slot](https://microbiomedata.github.io/nmdc-schema/depth/) already takes [objects](https://microbiomedata.github.io/nmdc-schema/QuantityValue/), as its values, with [has_minimum_numeric_value](https://microbiomedata.github.io/nmdc-schema/has_minimum_numeric_value/) and [has_maximum_numeric_value](https://microbiomedata.github.io/nmdc-schema/has_maximum_numeric_value/) sub-slots.
- when merging the `depth2` content into `depth`, should we just migrate the `depth`'s [has_minimum_numeric_value](https://microbiomedata.github.io/nmdc-schema/has_minimum_numeric_value/) value into the [has_minimum_numeric_value](https://microbiomedata.github.io/nmdc-schema/has_minimum_numeric_value/) slot if necessary?
- how do we determine that a biosample is a soil biosample?

**Comment by mslarae13** [comment]
With discussion during the squad sync, I realized "has_value" is the minimum value and the depth2 is the max value.

The metadata is there, but need to be more clearly stated "has minimum value"

**Comment by mslarae13** [comment]
@turbomam I think this is related to this. 
From the API, I see

`results__depth__has_raw_value`
- This is what gets entered into the submission portal, is a range for soil
`results__depth__has_minimum_numeric_value`
- The upper value
`results__depth__has_unit`
- the unit
`results__depth__has_maximum_numeric_value`
- the deepest value

And we WILL be keeping these 3 depth fields, yes? 

Is this issue resolved? Is depth2 gone? It looks like it is gone from the schema, but I am still seeing it in... runtime? and documentation? 
- https://github.com/search?q=org%3Amicrobiomedata+depth2&type=code&p=1

**Comment by turbomam** [comment]
Thanks for revisiting this issue. I don't completely understand your question, or which API method your talking about regarding  `results__depth__has_raw_value`. That GH code search is very helpful though! I'll look though that now.

As far as `nmdc-schema` or `berkeley-schema-fy24` are concerned, the range of `depth` is a one [QuantityValue](https://microbiomedata.github.io/nmdc-schema/QuantityValue/). We aren't requiring any of the `QuantityValue` slots in those schemas, but the intention is that a raw source value from the submission portal or some ETL process would go into `has_raw_value`, and some parsing process would determine whether the source value is a point or a range.

Depending on the template and our business logic, `has_raw_value` might be populated if the input is a point and we want to accept point/1D/single values. If we only want to retain ranges, then we would put that one value in both the `has_minimum_numeric_value' and `has_maximum_numeric_value` slots. And if the input is a range, then obviously we would do the same thing.

We will also need business logic in the ETLs from external sources or the Submission Portal to extract (or infer) a standardized unit string for `has_unit`.

I assume we would never populate `has_minimum_numeric_value' or `has_maximum_numeric_value` without its partner, and that we would never populate either of those along with `has_raw_value`.

`depth2`, like you have suggested, is not part of either schemas any more. It should be removed form all of those other repos too.

**Comment by turbomam** [comment]
repos mentioning `depth2`:
- `microbiomedata/DataHarmonizer`
    - I don't think we have any automation in which these files are processed any more.
    - I'm tempted to mark our fork of this repo as archived. I think @pkalita-lbl and I agreed on that a while ago. That wouldn't eliminate unhelpful search results like these `depth2` hits.
- `microbiomedata/NMDC_documentation`
    - The static schema content in this repo is ancient. I have raised that in several meetings (maybe Infrastructure meetings?) but I haven't identified anybody who wants to work with me on a  better integration between `nmdc-schema` and  `NMDC_documentation`. I think there were some discussion about just deleting the static content and providing links to the active schema documentation at https://microbiomedata.github.io/nmdc-schema/
- `microbiomedata/id_sandbox` looks like something I started and forgot about. I'll look more carefully after this.
-  `microbiomedata/metadata_converter` predates me. I could do dome research or we could wait for @cmungall to return form his vacation at the end of July.
- `microbiomedata/nmdc-runtime`
    - `depth2` certaihly shouldn't be mentioned in any documentation or tutorial like [this](https://github.com/microbiomedata/nmdc-runtime/blob/69bf65425e3ee0dc336a8b16080f583dccb3ce10/docs/tutorials/json.md?plain=1#L50)
    - I think the appearances of `depth2` could be removed from [docs/nb/get_data.ipynb](https://github.com/microbiomedata/nmdc-runtime/blob/69bf65425e3ee0dc336a8b16080f583dccb3ce10/docs/nb/get_data.ipynb#L156) by rerunning it, but it retrieves some records via retired `id`s, so we would have to work on that too.
    - I'm less certain bout some of the other files. Some look really old like [./tmp.json](https://github.com/microbiomedata/nmdc-runtime/blob/69bf65425e3ee0dc336a8b16080f583dccb3ce10/tmp.json#L7). Others may be illustration of how `depth2` recordes were mograted into the durrect `depth` model, and I would say that we should keep those.

**Comment by turbomam** [comment]
This was a really good exercise. I can't think of a way to catch things like this though automation, but when they come up for some other reason, it would be nice to take some corrective action.

**Comment by mslarae13** [comment]
@turbomam can you summarize and reiterate the goals of this issue & what has and hasn't been completed?

I think depth2 has been removed from the schema.. still some removal to do in other repos.

Sample type soil should be captured from the extension, but I'm not sure we kept a slot for that? If not, we should!

My question was just confirm that the slots are what I understand them to be and determining if the `depth2` issue is resolved. Trying to figure out what still needs done here.

**Comment by turbomam** [comment]
@mslarae13 can we discuss live either in tomorrow's _NMDC Schema & Metadata_ or before then if you're available?

**Comment by mslarae13** [comment]
depth2 hasn't been removed from all locations.. This kind of change needs some grooming.
Close this issue, but make a issue or discussion about development cycle of legacy things that don't get updated when the schema does.

Next steps to close this out
- Add depth2 to `deprecated.yaml` (deprecation?)
- Make issue for removing depth2 from other repos
- Make discussion around protocol for getting all downstream changes
- Make issue for adding test that fail when schema is changed
  - notebooks... leave it, you want to use it, you fix it.
  - Runtime should absolutely have tests and not break
- make an issue for public facing items should also have tests 
- make issue for notebook runner github action.

**Comment by ssarrafan** [comment]
> depth2 hasn't been removed from all locations.. This kind of change needs some grooming. Close this issue, but make a issue or discussion about development cycle of legacy things that don't get updated when the schema does.
> 
> Next steps to close this out
> 
> * Add depth2 to `deprecated.yaml` (deprecation?)
> * Make issue for removing depth2 from other repos
> * Make discussion around protocol for getting all downstream changes
> * Make issue for adding test that fail when schema is changed
>   
>   * notebooks... leave it, you want to use it, you fix it.
>   * Runtime should absolutely have tests and not break
> * make an issue for public facing items should also have tests
> * make issue for notebook runner github action.

Thanks @mslarae13 
Who will be making these other issues?  Let me know if I can help?
@turbomam

**Comment by mslarae13** [comment]
@turbomam to re-work this issue so the goal is more clear about QV vs min and max
